### PR TITLE
Add unit tests for SingleInstanceHelper and fix formatting from simplifying PR #11863

### LINKED ISSUE #12230: [Tracking] Current VB work (Refactoring and adding tests)
[Draft]

This will be the tracking issues for all the ongoing VB work that we want to accomplish in the .NET 10-time frame.

- [ ] https://github.com/dotnet/winforms/pull/12226 
(Refactoring out large code into 7 logical partial classes/code files)
- [ ] https://github.com/dotnet/winforms/pull/12221 
(Adding tests for VB App Framework Download tests)
- [ ] https://github.com/dotnet/winforms/pull/12227 
(Adding tests for VB App Framework Clipboard class)

### PR DESCRIPTION
**Pull request by paul1956**
Partially Fixes #12230

## Proposed changes
- Fix Formatting in SingleInstanceHelper code changes from #11863
- Add tests for SingleInstanceHelper from #11863

## Customer Impact
- Developer code cleanup and add tests

## Regression? 
- No

## Risk
- None

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/12261)

### CODE REVIEW

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/SingleInstanceHelpers.vb [hunk 1]
```diff
@@ -12,20 +12,26 @@ Namespace Microsoft.VisualBasic.ApplicationServices
...
+
             Using stream As New MemoryStream
                 While True
+                    Dim buffer As Byte() = New Byte(bufferLength - 1) {}
                     Dim bytesRead As Integer = Await pipeServer.ReadAsync(
-                        buffer.AsMemory(0, bufferLength),
-                        cancellationToken).ConfigureAwait(False)
+                        buffer:=buffer.AsMemory(start:=0, length:=bufferLength),
+                        cancellationToken) _
+                        .ConfigureAwait(continueOnCapturedContext:=False)
```
Comment (Tanya-Solyanik) [inline src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/SingleInstanceHelpers.vb:27]: Please indent by 4 spaces

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/SingleInstanceHelpers.vb [hunk 2]
```diff
@@ -12,20 +12,26 @@ Namespace Microsoft.VisualBasic.ApplicationServices
...
                     If bytesRead = 0 Then
                         Exit While
                     End If
                     Await stream.WriteAsync(
-                        buffer.AsMemory(0, bytesRead),
-                        cancellationToken).ConfigureAwait(False)
+                        buffer:=buffer.AsMemory(start:=0, length:=bytesRead),
+                        cancellationToken) _
+                        .ConfigureAwait(continueOnCapturedContext:=False)
```
Comment (Tanya-Solyanik) [inline src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/SingleInstanceHelpers.vb:34]: Please indent by 4 spaces

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/SingleInstanceHelpers.vb [hunk 3]
```diff
@@ -37,48 +43,70 @@ Namespace Microsoft.VisualBasic.ApplicationServices
...
             Dim content As Byte()
             Using stream As New MemoryStream
                 Dim serializer As New DataContractSerializer(GetType(String()))
                 serializer.WriteObject(stream, args)
                 content = stream.ToArray()
             End Using
-            Await pipeClient.WriteAsync(content.AsMemory(0, content.Length), cancellationToken).ConfigureAwait(False)
+            Await pipeClient.WriteAsync(
+                buffer:=content.AsMemory(start:=0, length:=content.Length), cancellationToken) _
+                .ConfigureAwait(continueOnCapturedContext:=False)
```
Comment (Tanya-Solyanik) [inline src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/SingleInstanceHelpers.vb:59]: Please indent by 4 spaces

File: src/Microsoft.VisualBasic.Forms/tests/UnitTests/System/Windows/Forms/SingleInstanceHelpersTests.vb [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        End Sub
+
+        <WinFormsFact>
+        Public Sub TryCreatePipeServerTwiceTests_Fail()
+            Dim pipeName As String = GetUniqueText()
+            Dim pipeServer As NamedPipeServerStream = Nothing
+            TryCreatePipeServer(pipeName, pipeServer).Should.BeTrue()
+            Dim pipeServer1 As NamedPipeServerStream = Nothing
+            TryCreatePipeServer(pipeName, pipeServer1).Should.BeFalse()
+            Using pipeServer
```
Comment (Tanya-Solyanik) [inline src/Microsoft.VisualBasic.Forms/tests/UnitTests/System/Windows/Forms/SingleInstanceHelpersTests.vb:43]: the `using` scope should start right after the IDisposable object is created, i.e. after TryCreatePipeServer

Comment (paul1956) [reply src/Microsoft.VisualBasic.Forms/tests/UnitTests/System/Windows/Forms/SingleInstanceHelpersTests.vb:41]: fied

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/SingleInstanceHelpers.vb (hunk 2 above)
Comment (paul1956) [reply src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/SingleInstanceHelpers.vb]: Fixed

File: src/Microsoft.VisualBasic.Forms/tests/UnitTests/System/Windows/Forms/SingleInstanceHelpersTests.vb [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+Namespace Microsoft.VisualBasic.Forms.Tests
+
+    Public Class SingleInstanceHelpersTests
+        Private _resultArgs As String()
+
+        Private Sub OnStartupNextInstanceMarshallingAdaptor(args As String())
+            If args.Length = 1 Then
+                _resultArgs = {"Hello"}
+            End If
+        End Sub
```
[the remark this answers is not in the document]
Comment (paul1956) [reply src/Microsoft.VisualBasic.Forms/tests/UnitTests/System/Windows/Forms/SingleInstanceHelpersTests.vb]: No way in VB that I know of to inline

[the remark this answers is not in the document]
Comment (paul1956) [reply src/Microsoft.VisualBasic.Forms/tests/UnitTests/System/Windows/Forms/SingleInstanceHelpersTests.vb]: Moved

**Pull request merged**